### PR TITLE
transformation adaptations 

### PR DESCRIPTION
**Pull request by waxenegger**
A general refactoring of transformations.

Uses omero-marshal schema 2016-06 for retrieval/storage with AffineTransform object (json).

Test: 
- Use http://web-dev-merge.openmicroscopy.org and open image with iviewer. 
- Note: shapes cannot (yet) be rotated in iviewer. 

1. Either choose an image that has alrady a shape with a transform (e.g. http://web-dev-merge.openmicroscopy.org/iviewer/9443/?dataset=1457 - user 1) or create a new one in insight.

2. Modify it and then store it. Open it again to see that is displays properly. Double check with the default web viewer and/or insight after persistence.

### CODE REVIEW

**Comment by jburel** [comment]
``
Conversion polygonToJsonObject:
     [exec] [0m[31m     the string "shape type polygon must be of instance ome.ol3.geom.Polygon!" was thrown, throw an Error :)[0m[90m
     [exec]   
zz

**Comment by will-moore** [comment]
Several rounds of rotate and save of Ellipses between Insight and iViewer and always displayed the same in both clients.
Good to merge.

File: ol3-viewer/src/ome/ol3/geom/Ellipse.js [hunk 1]
```diff
@@ -88,45 +85,19 @@ ome.ol3.geom.Ellipse = function(cx, cy, rx, ry, transform, theta, step) {
...
+    this.transform_ =
+        ome.ol3.utils.Transform.convertAffineTransformIntoMatrix(transform);
 
     /**
      * step the step size for plotting
      * @type {number}
      * @private
      */
-    this.step_ = opt['step'];
+    this.step_ = 0.1;
```
Comment (jburel) [inline ol3-viewer/src/ome/ol3/geom/Ellipse.js]: typo in the comment ``step the step``

**Comment by jburel** [comment]
Thanks Merging

**Pull request merged**